components/nodemanager-service/pgdb/filtering.go

```diff
@@ -245,14 +251,32 @@ func whereProjectsMatch(_ string, arr []string, tableAbbrev string) (string, err
 	var condition string
 
 	if len(arr) == 0 {
-		condition = fmt.Sprintf("not exists (select 1 from projects p join nodes_projects np on p.id = np.project_id where np.node_id = %s.id)", tableAbbrev)
-	} else {
-		value, err := pq.Array(arr).Value()
+		return "", nil
+	}
+
+	refinedValues := stringutils.SliceFilter(arr, func(projectId string) bool {
+		return projectId != authzConstants.UnassignedProjectID
+	})
+
+	if len(refinedValues) > 0 {
+		value, err := pq.Array(refinedValues).Value()
 		if err != nil {
 			return "", err
 		}
 		condition = fmt.Sprintf("exists (select 1 from projects p join nodes_projects np on p.id = np.project_id where np.node_id = %s.id and p.project_id = ANY('%s'::text[]))", tableAbbrev, value)
 	}
 
+	if stringutils.SliceContains(arr, authzConstants.UnassignedProjectID) {
+		unassignedCondition := fmt.Sprintf("not exists (select 1 from projects p join nodes_projects np on p.id = np.project_id where np.node_id = %s.id)", tableAbbrev)
+		if condition == "" {
+			condition = unassignedCondition
+		} else {
+			condition = condition + " OR " + unassignedCondition
```

Comment:
Consider using a query builder instead of using raw strings.
(Why? See my article [Code Smells: Raw Strings and the Subtle Syntax Catastrophe](https://www.red-gate.com/simple-talk/dotnet/net-development/code-smells-raw-strings-subtle-syntax-catastrophe/) )

I have been wanting to try out the [Squirrel query builder](https://github.com/Masterminds/squirrel) for some time, so hoping to pique your interest. So this line generates your alternation expression.
```
	condition, _, _ =
		sq.Or{
			sq.Expr(condition),
			sq.Expr(unassignedCondition),
		}.ToSql()
```
You can similarly build up more complex expressions like you have on L270.

Reply:
while an interesting thing to look into, that seems like the kinda thing worthy of its own pr, since we would want to thoroughly test that all the filtering still works as expected

Reply:
Thank you @msorens I have always used some kind of ORM before. With that library at least the SQL will be partially checked during compile time. 

@vjeffrey is right I would like to look into adding this type of work in a separate PR.